Article: how to track Vendée Globe race in GIS (#16)
geotribu/english-blog: content/posts/2024-12-02-vendee_globe_gis_data.md

---
authors:
  - ffougeres
categories:
  - Article
comments: true
date: 2024-12-02
description: Create and visualize GIS data on the progress of the Vendée Globe 2024 race from official spreadsheets.
icon: material/sail-boat

> license: beerware
> tags:

Reviewer: ```suggestion
license: beerware
links:
 - French version: https://geotribu.fr/articles/2024/2024-11-20_vendee_globe_donnees_sig/
 - Official race website: https://www.vendeeglobe.org/en
tags:
```

   - GeoPandas
   - Pandas
   - Python
   - QGIS
   - Vendée Globe
   - sailing

> title: Follow the Vendée Globe 2024 from a GIS

Reviewer: ```suggestion
title: Track the Vendée Globe race with Python and QGIS
```

> subtitle: Vendée Globe and GIS data

Reviewer: ```suggestion
subtitle: Sailing GIS
```

For downloading, I use the third-party library `requests`.

For reading the spreadsheet, cleaning the data and creating geometry, I use `pandas`, `geopandas` and `shapely`. There is some data cleaning to do, because the cells contain line breaks.

To go into more technical detail, once the file is downloaded, the successive steps are:

1. **Opening the file in a dataframe, keeping only the columns and rows that interest us.**  
This involves loading the Excel file and extracting it into a dataframe, keeping only the relevant data for further processing, while ignoring the superfluous information.

2. **Creating headers.**  
The headers of the Excel file are often made up of merged cells, which makes them difficult to retrieve. In addition, the column names are sometimes too verbose, so they need to be simplified to make them more usable.

3. **Data cleaning.**  
This step involves removing line breaks, special characters, or any other anomalies that could disrupt data processing.

>     ```pandas title="DataFrame before cleaning"
>    rang             code                                                nom         heure  ...   24h_vmg 24h_distance         dtf       dtl
>     0     1   GBR\r\nFRA 100                        Sam Goodchild\r\nVULNERABLE  10:30 FR\r\n  ...  10.5 kts     255.1 nm  22300.7 nm    0.0 nm
>     1     2   FRA\r\nFRA 112                 Sébastien Simon\r\nGroupe Dubreuil  10:30 FR\r\n  ...   7.4 kts     223.1 nm  22324.7 nm   24.0 nm
>     2     3    FRA\r\nFRA 59                        Thomas Ruyant\r\nVULNERABLE  10:30 FR\r\n  ...  10.7 kts     288.1 nm  22352.7 nm   52.0 nm
>     3     4     FRA\r\nFRA85                     Nicolas Lunven\r\nHOLCIM - PRB  10:30 FR\r\n  ...  12.7 kts     306.4 nm  22378.5 nm   77.8 nm
>     4     5    FRA\r\nFRA 29  Jean Le Cam\r\nTout commence en Finistère - Ar...  10:30 FR\r\n  ...   5.0 kts     158.5 nm  22379.0 nm   78.3 nm
>     5     6    FRA\r\nFRA 15          Clarisse Crémer\r\nL'Occitane en Provence  10:30 FR\r\n  ...   7.3 kts     211.9 nm  22410.7 nm  110.1 nm
>     ```

Reviewer: ````suggestion
    ```pandas title="DataFrame before cleaning"
        rang             code                                                nom         heure  ...   24h_vmg 24h_distance         dtf       dtl
    0     1   GBR\r\nFRA 100                        Sam Goodchild\r\nVULNERABLE  10:30 FR\r\n  ...  10.5 kts     255.1 nm  22300.7 nm    0.0 nm
    1     2   FRA\r\nFRA 112                 Sébastien Simon\r\nGroupe Dubreuil  10:30 FR\r\n  ...   7.4 kts     223.1 nm  22324.7 nm   24.0 nm
    2     3    FRA\r\nFRA 59                        Thomas Ruyant\r\nVULNERABLE  10:30 FR\r\n  ...  10.7 kts     288.1 nm  22352.7 nm   52.0 nm
    3     4     FRA\r\nFRA85                     Nicolas Lunven\r\nHOLCIM - PRB  10:30 FR\r\n  ...  12.7 kts     306.4 nm  22378.5 nm   77.8 nm
    4     5    FRA\r\nFRA 29  Jean Le Cam\r\nTout commence en Finistère - Ar...  10:30 FR\r\n  ...   5.0 kts     158.5 nm  22379.0 nm   78.3 nm
    5     6    FRA\r\nFRA 15          Clarisse Crémer\r\nL'Occitane en Provence  10:30 FR\r\n  ...   7.3 kts     211.9 nm  22410.7 nm  110.1 nm
    ```
````

---
authors:
  - ffougeres
categories:

>   - Announcements

Reviewer: ```suggestion
  - Article
```

comments: true
date: 2024-12-02
description: Create and visualize GIS data on the progress of the Vendée Globe 2024 race from official spreadsheets.
icon: material/sail-boat
license: beerware
tags:
   - GeoPandas
   - Pandas
   - Python
   - QGIS
   - Vendée Globe

>    - Sailing

Reviewer: ```suggestion
   - sailing
```

title: Follow the Vendée Globe 2024 from a GIS
subtitle: Vendée Globe and GIS data
---

# Follow the Vendée Globe 2024 from a GIS

> :calendar: Initial publication date: {{ page.meta.date | date_localized }}

Reviewer: ```suggestion
```

Not supported on English blog

```shell title="Format of the download URL of the scoreboard"
https://www.vendeeglobe.org/sites/default/files/ranking/vendeeglobe_leaderboard_YYYYMMDD_HHMMSS.xlsx
```

It is therefore necessary to complete the date (YYYYMMDD format) and the time of the score (HHMMSS) to construct the download URL.

Then, it is necessary to deal with the way in which the location data is presented. Indeed, the positions of the boats are often provided in a geographic coordinate format in degrees, minutes and seconds (DMS). Although this format is useful, it is not directly compatible with geomatics tools. It is therefore essential to convert them into decimal degrees, a more standard and precise format, which allows easy working with maps and geographic information systems (GIS).

Finally, it is important to export this GIS data in a compatible format, such as GeoPackage or GeoJSON. Once converted, this data can be used in any GIS, whether it is a desktop GIS like QGIS or a web GIS map with tools like MapLibre or Leaflet.

----

## Industrialize the method

> ! [factory logo](https://cdn.geotribu.fr/img/logos-icones/divers/factory.png){: .img-thumbnail-left }

Reviewer: broken syntax

```suggestion
![factory logo](https://cdn.geotribu.fr/img/logos-icones/divers/factory.png){: .img-thumbnail-left }
```

    "skipper" = 'Maxime Sorel'
    ```
<!-- markdownlint-enable MD046 -->

## To go further

This first step is only a POC (Proof of Concept) the code can still be optimized and I will continue to do so throughout the race (hoping that the formalism and publication times of the spreadsheet do not change). Subsequently, several ideas could be explored. I will surely explore one of them.

- **Create a QGIS plugin**: A QGIS plugin could allow to load the ranking, the last position of the ships, and their trajectory. We could imagine that the post-processing of the Excel file to GIS data is done by continuous integration (CI) and exported in GeoJSON, and that the plugin loads these GeoJSON hosted in the GitHub project.

- **Provide data via an API**: We could imagine a project that automatically retrieves this data, converts and structures it, then exposes an API that provides a position or trajectory based on a competitor's number, for example.

- **Create a cartographic web application** to visualize the progress of the boats with more possibilities than what the official cartographic interface offers. I had imagined using [mviewer](https://mviewer.github.io/fr/) for this.

> ----
> 
> <!-- geotribu:authors-block -->
> 
> {% include "licenses/beerware.md" %}

Reviewer: ```suggestion
```

Not necessary on English blog
